src/shared/redux/slices/settingsSlice.ts

```diff
@@ -60,7 +44,20 @@ export const selectDefaultDirPath = (state: RootState) =>
 export const selectCurrentLocale = (state: RootState) =>
   state.appSettings.currentLocale;
 
-export const selectVersionDetails = (state: RootState) =>
-  state.appSettings.versionDetails;
+export const saveSettingsAsync = createAsyncActionMain<Partial<Settings>>(
+  'saveSettings',
+  (settings) => async (dispatch) => {
+    dispatch(setSettings(settings));
+    writeJSON(SETTINGS_FILE_PATH, settings);
+  }
```

Comment:
@LostInStatic this will save the partial to the json file. In case of settings view it works all goof but in case of the lang changes on log in screen it will save only the `currentLocale` to the json.

![persist](https://user-images.githubusercontent.com/34404855/143405602-c6670380-4b8a-4cdb-9a01-20460850f5f6.gif)

Perhaps using the `getState` can be of help here:

```ts
export const saveSettingsAsync = createAsyncActionMain<Partial<Settings>>(
  'saveSettings',
  (settings) => async (dispatch, getState) => {
    const nextSettings = { ...getState().appSettings, ...settings };
    dispatch(setSettings(nextSettings));
    writeJSON(SETTINGS_FILE_PATH, nextSettings);
  }
);
```

Reply:
Right... 🤦‍♂️ Fixed.